### PR TITLE
Remove nivida/apex

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title

### Does your PR introduce any breaking changes? If yes, please list them.

Removes apex support

cc @justusschock @awaelchli @borda @carmocca

### CODE REVIEW

File: src/pytorch_lightning/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,20 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 
+
+## [fabricLaunchPLVersion] - 202Y-MM-DD
+
+- `nvidia/apex` removal
```
Comment (carmocca) [inline src/pytorch_lightning/CHANGELOG.md:11]:
```suggestion
- `nvidia/apex` removal ([#16149](https://github.com/Lightning-AI/lightning/pull/16149))
```

File: src/pytorch_lightning/trainer/connectors/checkpoint_connector.py [hunk 2]
```diff
@@ -293,8 +293,8 @@ def restore_precision_plugin_state(self) -> None:
             prec_plugin.load_state_dict(self._loaded_checkpoint[prec_plugin.__class__.__qualname__])
 
         # old checkpoints compatibility
-        if "amp_scaling_state" in self._loaded_checkpoint and isinstance(prec_plugin, ApexMixedPrecisionPlugin):
-            prec_plugin.load_state_dict(self._loaded_checkpoint["amp_scaling_state"])
+        if "amp_scaling_state" in self._loaded_checkpoint:
+            rank_zero_warn("This checkpoint contains apex AMP data, but apex support has been removed.")
```
Comment (awaelchli) [inline src/pytorch_lightning/trainer/connectors/checkpoint_connector.py:297]: Should we remove this as well?

**Review by awaelchli — APPROVED** [review]

File: src/pytorch_lightning/trainer/connectors/checkpoint_connector.py (hunk 2 above)
Comment (carmocca) [reply src/pytorch_lightning/trainer/connectors/checkpoint_connector.py:297]: Fine with me either way. I thought having a warning could be useful

Comment (awaelchli) [reply src/pytorch_lightning/trainer/connectors/checkpoint_connector.py:297]: Alternatively we could also have a migration that removes the key and warns that this state gets dropped.

**Review by akihironitta — APPROVED** [review]

File: src/pytorch_lightning/trainer/connectors/checkpoint_connector.py (hunk 2 above)
Comment (carmocca) [reply src/pytorch_lightning/trainer/connectors/checkpoint_connector.py:297]: Leaving this for a follow-up

**Pull request merged**